src/GovUk.Education.ExploreEducationStatistics.Content.Model.Tests/Repository/MethodologyVersionRepositoryTests.cs

```diff
@@ -854,6 +854,152 @@ await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId)
+
+            await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId))
+            {
+                var service = BuildMethodologyVersionRepository(
+                    contentDbContext: contentDbContext);
+
+                var result = await service
+                    .GetLatestPublishedVersionBySlug("slug");
+
+                VerifyAllMocks(methodologyRepository);
+
+                Assert.NotNull(result);
+                Assert.Equal(latestPublishedVersionId, result.Id);
+                Assert.Equal("slug", result.Slug);
+            }
+        }
+
+        [Fact]
+        public async Task GetLatestPublishedVersionBySlug_OwningPublicationSlug()
+        {
+            var latestPublishedVersionId = Guid.NewGuid();
+            var methodology = new Methodology
+            {
+                LatestPublishedVersionId = latestPublishedVersionId,
+                OwningPublicationSlug = "slug",
+                Versions = new List<MethodologyVersion>
+                {
+                    new()
+                    {
+                        AlternativeSlug = "not-like-this",
+                        Version = 1,
+                        PreviousVersionId = latestPublishedVersionId,
+                    },
+                    new()
+                    {
+                        Id = latestPublishedVersionId,
+                        AlternativeSlug = null,
+                        Version = 0,
+                    },
+                },
+            };
+
+            var contentDbContextId = Guid.NewGuid().ToString();
+            await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId))
+            {
+                await contentDbContext.Methodologies.AddAsync(methodology);
+                await contentDbContext.SaveChangesAsync();
+            }
+
+            var methodologyRepository = new Mock<IMethodologyRepository>(Strict);
+
+            await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId))
+            {
+                var service = BuildMethodologyVersionRepository(
+                    contentDbContext: contentDbContext);
+
+                var result = await service
+                    .GetLatestPublishedVersionBySlug("slug");
+
+                VerifyAllMocks(methodologyRepository);
+
+                Assert.NotNull(result);
+                Assert.Equal(latestPublishedVersionId, result.Id);
+                Assert.Null(result.AlternativeSlug);
+                // doesn't return result.Methodology, so cannot check result.Slug/result.Methodology.OwningPublicationSlug
+            }
+        }
+
+        [Fact]
+        public async Task GetLatestPublishedVersionBySlug_UnpublishedMethodology()
+        {
+            var methodology = new Methodology
+            {
+                LatestPublishedVersionId = null,
+                OwningPublicationSlug = "slug",
```

Comment:
All of these being "slug", I'm not 100% what we're proving here :)  Do any of these need changing to provide more proof that the mechanism is working or are you happy it's OK as it is?

Reply:
Actually I see the result is null, but I'm guessing that in real life this data setup is not possible as we'd need to publish the first version before creating the second version?

Reply:
I've updated the test data so it's more realistic